### PR TITLE
refactor: resolve the `result_large_err` clippy lint warnings by shrinking core error types where appropriate

### PR DESCRIPTION
**Pull request by cylewitruk-stacks**
## Summary

Resolves the remaining `clippy::result_large_err` warnings by reducing the size of several error types and refactors some tuple-types in the refactor path to use concrete types. This does not change runtime, RPC, or wire-format behavior.

## Changes

- Box large Clarity parsing and static-analysis errors.
- Box Bitcoin message and StackerDB acknowledgement payloads.
- Replace nested error tuples with named contexts for cost overflows, nonce checks, and microblock processing.
- Keep `VmExecutionError` inline because it does not require boxing.
- Box one RPC-local `NetError` to keep its enclosing tuple below the lint threshold.

The affected consensus snapshots now reflect the named `CostOverflowContext { before, after, budget }` debug output.

Regenerating them also exposed stale block and MARF hashes in one ignored static-analysis snapshot. Those hashes predated the PoX-5 test-harness changes and were identical across two fresh runs, so they are refreshed here as well (`chainstate::tests::static_analysis_tests::static_check_error_cost_balance_exceeded`).

⚠️ These changes affect public Rust error shapes, so downstream consumers may need to update pattern matches and affected helper call sites (e.g. Clarinet).

### CODE REVIEW

File: changelog.d/reduce-large-error-types.changed [hunk 1]
```diff
@@ -0,0 +1 @@
+Shrinked a number of root error types through `Box`ing of large variants to satisfy the `clippy::result_large_err` lint.
```
[the remark this answers is not in the document]
Comment (cylewitruk-stacks) [reply changelog.d/reduce-large-error-types.changed]: 🤦 [e8552a4](https://github.com/stacks-network/stacks-core/pull/7574/commits/e8552a4a7447295957bc9fc4d1718e98227674cd)

[the remark this answers is not in the document]
Comment (cylewitruk-stacks) [reply changelog.d/reduce-large-error-types.changed]: Yeah, fair, changed to breaking in [e8552a4](https://github.com/stacks-network/stacks-core/pull/7574/commits/e8552a4a7447295957bc9fc4d1718e98227674cd).

File: changelog.d/7574-reduce-large-error-types.changed [hunk 2]
```diff
@@ -0,0 +1 @@
+Shrank a number of root error types through `Box`ing of large variants to satisfy the `clippy::result_large_err` lint, which may be API-breaking for some downstream library consumers.
```
[the remark this answers is not in the document]
Comment (cylewitruk-stacks) [reply changelog.d/7574-reduce-large-error-types.changed:1]: I think this is probably overemphasizing the breaking nature of this change; "breaking" in this context, I believe, is more for true "network operator must do something to upgrade"-style changes (i.e. breaking config change).

File: stackslib/src/chainstate/tests/snapshots/blockstack_lib__chainstate__tests__static_analysis_tests__static_check_error_cost_balance_exceeded.snap
Comment (francesco-stacks) [inline stackslib/src/chainstate/tests/snapshots/blockstack_lib__chainstate__tests__static_analysis_tests__static_check_error_cost_balance_exceeded.snap:1]: sanity check: in this snapshot the block and marf hashes are actually different. Is it possible that this change introduced a consensus breaking change for an edge-case? or maybe it's just that these snapshots were not updated (maybe the tests were ignored by default?)

Comment (cylewitruk-stacks) [reply stackslib/src/chainstate/tests/snapshots/blockstack_lib__chainstate__tests__static_analysis_tests__static_check_error_cost_balance_exceeded.snap:1]: Yeah my take is that these were ignored and not run as part of the unit test matrix by default and thus the previous Pox5 harness changes which caused the diff went unnoticed. That's what the comment in the description is trying to allude to:

>Regenerating them also exposed stale block and MARF hashes in one ignored static-analysis snapshot. Those hashes predated the PoX-5 test-harness changes and were identical across two fresh runs, so they are refreshed here as well (chainstate::tests::static_analysis_tests::static_check_error_cost_balance_exceeded).

**Review by francesco-stacks — DISMISSED** [review]

**Review by francesco-stacks — APPROVED** [review]

**Review by federico-stacks — APPROVED** [review]
